src/wasm-interpreter.h

```diff
@@ -1253,12 +1256,60 @@ class ModuleInstanceBase {
     }
     Flow visitMemoryCopy(MemoryCopy *curr) {
       NOTE_ENTER("MemoryCopy");
-      // TODO(tlively): implement me
+      Flow dest = this->visit(curr->dest);
+      if (dest.breaking()) return dest;
+      Flow source = this->visit(curr->source);
+      if (source.breaking()) return source;
+      Flow size = this->visit(curr->size);
+      if (size.breaking()) return size;
+      NOTE_EVAL1(dest);
+      NOTE_EVAL1(source);
+      NOTE_EVAL1(size);
+      Address destVal(uint32_t(dest.value.geti32()));
+      Address sourceVal(uint32_t(source.value.geti32()));
+      Address sizeVal(uint32_t(size.value.geti32()));
+
+      instance.checkLoadAddress(destVal, 0);
```

Comment:
This doesn't seem right to me -- the bulk memory proposal requires that a copy that would trap is partially written, up to the point that the trap occurs.

Reply:
Yes, but this line is checking that the destination address is in bounds at all (or one past the end, as we discussed earlier). The bounds checking for each individual byte is done by `getFinalAddress` in the loop below.

Reply:
Oh, does that throw an exception? If so that works too.